### PR TITLE
Update typescript definition of toIdSchema(), and passed the new `idSeparator` prop through to the `AnyOfField` and `OneOfField` inside of `SchemaField`

### PR DESCRIPTION
**Pull request by heath-freenome**
### Reasons for making this change
Follow-up fix to #2628 

- Add the new `idSeparator` prop to the `index.d.ts` file for the `toIdSchema()` function definition
- Also fixed the spelling of `idPrefix` parameter
- Also passed `idSeparator` through to the `AnyOfField` and `OneOfField` in the `SchemaField` code
- Updated `ArrayField` to also pass `idPrefix` and `idSeparator` down through to the inner `SchemaField` used for array field items
  - This includes adding a test to validate that it is properly being passed down now

### Checklist

* [ ] **I'm updating documentation**
  - [ ] I've [checked the rendering](https://react-jsonschema-form.readthedocs.io/en/latest/#contributing) of the Markdown text I've added
* [ ] **I'm adding or updating code**
  - [ ] I've added and/or updated tests
  - [ ] I've updated [docs](https://react-jsonschema-form.readthedocs.io/) if needed
  - [ ] I've updated the [changelog](https://github.com/rjsf-team/react-jsonschema-form/blob/master/CHANGELOG.md) with a description of the PR
* [ ] **I'm adding a new feature**
  - [ ] I've updated the playground with an example use of the feature

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -17,11 +17,14 @@ should change the heading of the (upcoming) version to include a major version b
...
 
-# v4.0.2 (upcoming)
+# v4.0.3 (upcoming)
+
+# v4.0.2
 
 ## @rjsf/core
 
 - To improve performance, skip validating subschemas in oneOf / anyOf if formData is undefined (#2676)
+- Fixed the definition of the `toIdSchema()` typescript definition to add new `idSeparator` prop
```
Comment (epicfaace) [inline CHANGELOG.md:27]:
```suggestion
- Fixed the `toIdSchema()` typescript definition to add new `idSeparator` prop
```

**Review by epicfaace — APPROVED** [review]

**Review by epicfaace — CHANGES_REQUESTED** [review]
Can you update the PR description / changelog as per this latest change?

**Comment by heath-freenome** [comment]
> Do we need to update the documentation at all (maybe where the AnyOfField / OneOfField props are documented)?

I just scanned the docs for `idPrefix` (the sibling to `idSeparator`) and it isn't mentioned on those two fields. Actually, `AnyOfField` and `OneOfField` are only mentioned once, it the place where it describes fields you can override

File: packages/core/src/components/fields/ArrayField.js [hunk 2]
```diff
@@ -819,6 +819,8 @@ class ArrayField extends Component {
           uiSchema={itemUiSchema}
           formData={itemData}
           errorSchema={itemErrorSchema}
+          idPrefix={this.props.idPrefix}
```
[the remark this answers is not in the document]
Comment (heath-freenome) [reply packages/core/src/components/fields/ArrayField.js:822]: I think it is closer to a bug fix... I can add a test

[the remark this answers is not in the document]
Comment (heath-freenome) [reply packages/core/src/components/fields/ArrayField.js:822]: @epicfaace Test pushed... Did you want to reapprove or shall I merge?

File: CHANGELOG.md [hunk 3]
```diff
@@ -17,11 +17,16 @@ should change the heading of the (upcoming) version to include a major version b
...
+# v4.0.3 (upcoming)
+
+# v4.0.2
 
 ## @rjsf/core
 
 - To improve performance, skip validating subschemas in oneOf / anyOf if formData is undefined (#2676)
+- Fixed the `toIdSchema()` typescript definition to add new `idSeparator` prop
+  - Also passed the new `idSeparator` prop through to the `AnyOfField` and `OneOfField` inside of `SchemaField` 
+  - Updated `ArrayField` in `@rjsf/core` to pass `idSeparator` and `idPrefix` through to `SchemaField`
```
Comment (epicfaace) [inline CHANGELOG.md:29]: Can you update this line to make it clearer that a bug was fixed?

File: CHANGELOG.md [hunk 4]
```diff
@@ -17,11 +17,16 @@ should change the heading of the (upcoming) version to include a major version b
...
```
Comment (epicfaace) [inline CHANGELOG.md:30]: Can you also update the PR title to note the 3 things that were changed?

File: CHANGELOG.md (hunk 3 above)
Comment (epicfaace) [reply CHANGELOG.md]: @heath-freenome I think you forgot to do this

Comment (heath-freenome) [reply CHANGELOG.md]: > @heath-freenome I think you forgot to do this

It is there... the `idPrefix={this.props.idPrefix}` and `idSeparator={this.props.idSeparator}`

Comment (epicfaace) [reply CHANGELOG.md]: I meant, "update this line to make it clearer that a bug was fixed"

**Pull request merged**